File: SIAP/src/main/java/com/ryuseicode/siap/service/requisition/imp/ChapterService.java
package com.ryuseicode.siap.service.requisition.imp;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.ryuseicode.siap.entity.requisition.Chapter;
import com.ryuseicode.siap.repository.requisition.imp.ChapterRepository;
import com.ryuseicode.siap.service.requisition.intf.IChapterService;

/**
 * @name ChapterService
 * {@summary Service class to implement the behavior of IChapterService }
 * @author Ricardo Sanchez Romero (ricardo.sanchez@ryuseicode.com)
 * @since Dec 22, 2019
 */
@Service
public class ChapterService implements IChapterService {
	/**
	 * ChapterRepository 
	 */
	@Autowired
	private ChapterRepository chapterRepository;
	/**
	 * @name GetByBudgetId
	 * {@summary Method to get by budgetId }
	 * @param budgetId
	 * @return
	 */
	public List<Chapter> getByBudgetId(int budgetId) {
		return this.chapterRepository.getByBudgetId(budgetId);
	}
	/**
	 * @name GetByChapterId
	 * {@summary Method to get by chapterId }
	 * @param chapterId
	 * @return
	 */
	public Chapter getByChapterId(int chapterId) {
		return this.chapterRepository.getByChapterId(chapterId);
	}
}

File: SIAP/src/main/resources/static/Elysium/App/Services/Requisition/ChapterService.js
/**
 * @name ChapterService
 * @abstract Service class
 * @author Ricardo Sanchez Romero (ricardo.sanchez@ryuseicode.com)
 * @CreationDate 2019-11-30
 */
Elysium.App.Services.Requisition.ChapterService = (function () {
    /**********************************************/
    /*                   Variables                */
    /**********************************************/
    var Api = Elysium.AppHost + 'requisition/chapter';
    /**********************************************/
    /*                   Methods                  */
    /**********************************************/
    /**
     * @name GetByBudgetId
     * @abstract Method to get a collection of objects
     */
    var GetByBudgetId = function (budgetId) {
        // Define promise
        var deferred = new $.Deferred();
        // Check Access        
        var url = Api + '/getByBudgetId/' + budgetId;
        $.ajax({
            // tipo de llamado
            type: 'GET',
            // url del llamado a ajax
            url: url,
            // obtener siempre los datos mas actualizados
            cache: false,
            // tipo de datos de regreso
            dataType: "json",
            // funcion en caso de exito
            success: function (data, textStatus, jqXHR) {
                deferred.resolve(data);
            },
            // funcion en caso de error
            error: function (xhr, status, error) {
                deferred.reject(xhr);
            }
        });
        // return promise
        return deferred.promise();
    };
    /**
     * @name GetByChapterId
     * @abstract Method to get by chapter id
     * 
     */
    var GetByChapterId = function (chapterId) {
        // Define promise
        var deferred = new $.Deferred();
        // Check Access        
        var url = Api + '/getByChapterId/' + chapterId;
        $.ajax({
            // tipo de llamado
            type: 'GET',
            // url del llamado a ajax
            url: url,
            // obtener siempre los datos mas actualizados
            cache: false,
            // tipo de datos de regreso
            dataType: "json",
            // funcion en caso de exito
            success: function (data, textStatus, jqXHR) {
                deferred.resolve(data);
            },
            // funcion en caso de error
            error: function (xhr, status, error) {
                deferred.reject(xhr);
            }
        });
        // return promise
        return deferred.promise();
    };
    /**********************************************/
    /*                 Encapsulate                */
    /**********************************************/
    return {
    	GetByBudgetId: GetByBudgetId,
    	GetByChapterId: GetByChapterId
    };
}());